src/overlays/actors/ovl_Bg_Bowl_Wall/z_bg_bowl_wall.c

```diff
@@ -33,21 +40,188 @@ const ActorInit Bg_Bowl_Wall_InitVars = {
+            effectPos.z = Math_Rand_CenteredFloat(400.0f) + this->bullseyeCenter.z;
+            EffectSsBomb2_SpawnLayered(globalCtx, &effectPos, &effectVelocity, &effectAccel, 100, 30);
+            effectPos.y = -50.0f;
+            EffectSsHahen_SpawnBurst(globalCtx, &effectPos, 10.0f, 0, 50, 15, 3, HAHEN_OBJECT_DEFAULT, 10, NULL);
+            Audio_PlayActorSound2(&this->dyna.actor, NA_SE_IT_BOMB_EXPLOSION);
+        }
+        quakeIndex = Quake_Add(globalCtx->cameraPtrs[globalCtx->activeCamera], 1);
+        Quake_SetSpeed(quakeIndex, 0x7FFF);
+        Quake_SetQuakeValues(quakeIndex, 300, 0, 0, 0);
+        Quake_SetCountdown(quakeIndex, 30);
+        this->timer = 20;
+        this->actionFunc = BgBowlWall_FinishFall;
+    }
+}
+
+void BgBowlWall_FinishFall(BgBowlWall* this, GlobalContext* globalCtx) {
+    if (this->timer >= 2) {
+        if (this->dyna.actor.params == 0) {
+            Math_SmoothScaleMaxMinS(&this->dyna.actor.shape.rot.x, -0x3E80, 1, 200, 0);
+        } else {
+            Math_SmoothScaleMaxF(&this->dyna.actor.posRot.pos.y, this->initPos.y - 450.0f, 0.3f, 10.0f);
+        }
+    } else if (this->timer == 1) {
+        this->dyna.actor.posRot.rot.x = this->dyna.actor.shape.rot.x = 0;
+        this->dyna.actor.posRot.pos.y = this->initPos.y - 450.0f;
+        this->chuGirl->wallStatus[this->dyna.actor.params] = 2;
+        this->actionFunc = BgBowlWall_Reset;
+    }
+}
+
+void BgBowlWall_Reset(BgBowlWall* this, GlobalContext* globalCtx) {
+    if (this->chuGirl->wallStatus[this->dyna.actor.params] != 2) {
+        Math_SmoothScaleMaxF(&this->dyna.actor.posRot.pos.y, this->initPos.y, 0.3f, 50.0f);
+        if (fabsf(this->dyna.actor.posRot.pos.y - this->initPos.y) <= 10.0f) {
+            this->dyna.actor.posRot.pos.y = this->initPos.y;
+            this->isHit = false;
+            this->actionFunc = BgBowlWall_SpawnBullseyes;
+        }
+    }
+}
+
+void BgBowlWall_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgBowlWall* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Bowl_Wall/func_8086F260.s")
+    if (this->timer != 0) {
+        this->timer--;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Bowl_Wall/func_8086F440.s")
+    this->actionFunc(this, globalCtx);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Bowl_Wall/func_8086F464.s")
+void BgBowlWall_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    GlobalContext* globalCtx2 = globalCtx;
+    BgBowlWall* this = THIS;
+    u32 frames;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Bowl_Wall/func_8086F718.s")
+    OPEN_DISPS(globalCtx2->state.gfxCtx, "../z_bg_bowl_wall.c", 441);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Bowl_Wall/func_8086F7F8.s")
+    func_80093D84(globalCtx2->state.gfxCtx);
+    gSPSegment(POLY_OPA_DISP++, 8,
+               Gfx_TexScroll(globalCtx2->state.gfxCtx, 0, -2 * (frames = globalCtx2->state.frames), 16, 16));
```

Comment:
Any chance this assignment can be done separately?

Reply:
messed with it a bit just now, cant get anything to work. tried some usual tricks like * 1 and moving the negative sign and stuff